go.mod

```diff
@@ -2,16 +2,18 @@ module github.com/grafana/grafana
 
 go 1.23.1
 
-// contains openapi encoder fixes. remove ASAP
-replace cuelang.org/go => github.com/grafana/cue v0.0.0-20230926092038-971951014e3f // @grafana/grafana-as-code
+// Override Prometheus version because Prometheus v2.X is tagged as v0.X for Go modules purposes and Go assumes
+// that v1.Y is higher than v0.X, so when we resolve dependencies if any dependency imports v1.Y we'd
+// import that instead of v0.X even though v0.X is newer.
+replace github.com/prometheus/prometheus => github.com/prometheus/prometheus v0.52.0
```

Comment:
This was removed explicitly in this PR https://github.com/grafana/grafana/pull/96925

Reply:
I'm gonna handle it in this PR https://github.com/grafana/grafana/pull/98848